R/test-threads.R

```diff
@@ -330,4 +298,63 @@ delete_models <- function(.mods, .tags = "test threads", .force = FALSE){
 }
 
 
+#' Parse model files and overwrite estimation iterations
+#'
+#' @param .mods a bbi model object or list of model objects. Generally created by `test_threads()`.
+#' @param .max_eval Max number of iterations for NONMEM to run.
+#' @param .detect Estimation options that will be set to `.max_eval` if found.
+#'
+#' @importFrom readr parse_number
+#'
+#' @keywords internal
+adjust_estimation_options <- function(.mods, .max_eval, .detect = c("MAXEVAL", "NITER", "NBURN", "MAX")){
+
+  assert_true(all(.detect %in% c("MAXEVAL", "NITER", "NBURN", "MAX")))
```

Comment:
Not entirely sure if the assert call is necessary, but thoughts were that we may eventually want to specify `.detect` in the function call, in which case it would be helpful to check that they were typed in correctly. The new test function was set up similarly